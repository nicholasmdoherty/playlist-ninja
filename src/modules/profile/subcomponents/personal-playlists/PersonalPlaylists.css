
.playlists-container {
  display: flex;
  flex-direction: column;
  gap: 10px;
  width: 100%;
}

.playlists-header {
  font-family: 'Roc Grotesk';
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 35px;
}

.playlists-nav-row {
  display: flex;
  gap: 10px;
}

.playlists-card {
  display: flex;
  flex-direction: column;
  overflow: hidden;
}

.playlists-pagination {
  display: flex;
  gap: 10px;
  align-items: center;
  background: #353535;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.25);
  padding: 10px 15px;
  border-radius: 15px;
  font-size: 14px;
}

.playlists-pagination-arrow:hover {
  color: #09C053;
  cursor: pointer;
}

.playlists-pagination-arrow.disabled {
  color: #8C8C8C;
}

.playlists-search {
  display: flex;
  gap: 10px;
  align-items: center;
  background: #353535;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.25);
  padding: 10px 15px;
  border-radius: 15px;
  font-size: 14px;
  flex-grow: 1;
}

.playlists-sort {
  display: flex;
  gap: 10px;
  align-items: center;
  background: #353535;
  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.25);
  padding: 10px 15px;
  border-radius: 15px;
  font-size: 14px;
}

.playlist-row {
  display: flex;
  flex-direction: row;
  padding: 15px;
  justify-content: space-between;
  align-items: center;
  gap: 20px;
}

.playlist-row.odd {
  background-color: #444444;
}

.playlist-image {
  border-radius: 5px;
}

.playlist-img-name {
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-items: center;
}

.playlist-name {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;  
  overflow: hidden;
  line-clamp: 2;
  text-overflow: ellipsis;
  line-height: 18px;
  font-size: 16px;
  max-width: 100%;
}

.edit-button {
  display: flex;
  gap: 5px;
  align-items: center;
  padding: 10px 15px;
  border-radius: 15px;
  font-size: 14px;
  background: white;
  color: black;
  width: fit-content;
}

.edit-button:hover {
  cursor: pointer;
  background: linear-gradient(rgba(0,0,0, .2), rgba(0,0,0,.2)), white;
  text-decoration: underline;
}