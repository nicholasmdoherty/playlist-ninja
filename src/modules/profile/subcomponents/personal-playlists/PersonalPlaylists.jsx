import React, { Component } from "react";
import { Row, Col } from "react-bootstrap";
import { connect } from "react-redux";
import autoBind from "react-autobind";
import PlaylistCard from "./PlaylistCard";
import Spacer from "../../../../common/components/Spacer";
import { loadUsersEditablePlaylists } from "../../../../redux/actions/playlistActions";

class PersonalPlaylists extends Component {
  constructor(props) {
    super(props);

    this.state = {
      playlists: null,
      error: null
    };

    autoBind(this);
  }

  componentDidMount() {
    this.getPersonalPlaylistData();
  }

  getPersonalPlaylistData() {
    let { userId, api } = this.props;

    if (userId) {
      this.props.loadEditablePlaylists(userId, api);
    } else {
      this.setState({ error: "No user ID found, please reload" });
    }
  }

  render() {
    if (this.props.playlists) {
      let { playlists } = this.props;

      return (
        <Row className="text-center m-3 pb-2">
          <Col xs={12}>
            <h3 className="display-5"> My Playlists </h3>
            <Spacer percentage={3} />
          </Col>

          {playlists.map(playlist => {
            return (
              <PlaylistCard
                playlist={playlist}
                api={this.props.api}
                updatePlaylists={this.getPersonalPlaylistData}
              />
            );
          })}
        </Row>
      );
    } else {
      return null;
    }
  }
}

export const mapStateToProps = state => {
  return {
    api: state.api.spotifyApi,
    playlists: state.playlist.editablePlaylists
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    loadEditablePlaylists: (userId, spotifyApi) =>
      dispatch(loadUsersEditablePlaylists(userId, spotifyApi))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PersonalPlaylists);
