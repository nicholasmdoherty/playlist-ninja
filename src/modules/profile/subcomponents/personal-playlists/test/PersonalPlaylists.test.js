// Package imports
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

// Project Imports
import PersonalPlaylists from "../PersonalPlaylists";
import configureStore from "../../../../../store";

describe("PersonalPlaylists.jsx", () => {
  it("matches the snapshot for empty props", () => {
    const component = renderer.create(
      <Provider store={configureStore()}>
        <PersonalPlaylists />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
