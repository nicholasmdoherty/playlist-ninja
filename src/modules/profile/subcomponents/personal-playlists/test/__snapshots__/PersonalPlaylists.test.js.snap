// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalPlaylists.jsx matches the snapshot for empty props 1`] = `null`;
